File: testclient/src/main/java/com/example/testclient/PermissionUtils.java
package com.example.testclient;

import android.Manifest;
import android.app.Activity;

import androidx.core.app.ActivityCompat;

/**
 * Created by shqyang on 04/18/2021
 */
public class PermissionUtils {

    /**
     * request permission to read file
     * @param act Activity
     * @param requestCode request code
     */
    public static boolean requestReadFile(Activity act, int requestCode) {
        if (ActivityCompat.checkSelfPermission(act, Manifest.permission.READ_EXTERNAL_STORAGE) == 0) {
            return true;
        }
        ActivityCompat.requestPermissions(act, new String[] {
                Manifest.permission.READ_EXTERNAL_STORAGE
        }, requestCode);
        return false;
    }

    public static boolean hasReadFile(Activity act) {
        return ActivityCompat.checkSelfPermission(act, Manifest.permission.READ_EXTERNAL_STORAGE) == 0;
    }

}

File: app/src/main/java/com/sw/haruka/helper/utils/PermissionUtils.kt
package com.sw.haruka.helper.utils

import android.app.Activity
import androidx.core.app.ActivityCompat

/**
 * Created by Swsbty on 2021/03/14
 */
object PermissionUtils {

    const val REQUEST_PERMISSION_CODE = 9527
    fun file(act: Activity): Boolean {
        if (ActivityCompat.checkSelfPermission(act, android.Manifest.permission.WRITE_EXTERNAL_STORAGE) != 0) {
            ActivityCompat.requestPermissions(act, arrayOf(android.Manifest.permission.WRITE_EXTERNAL_STORAGE,
                android.Manifest.permission.READ_EXTERNAL_STORAGE), REQUEST_PERMISSION_CODE)
            return false
        }
        return true
    }

}